src/app/components/form-fields/date-time-field/date-time-field.component.ts

```diff
@@ -29,6 +41,43 @@ export class DateTimeFieldComponent extends BaseDirectFormField<Date, [DateTimeF
         return true;
     };
 
+    public validTimeChange(event: boolean): void {
+        this.isTimeValid = event;
+        this.onValueChange(event);
+    }
+
+    public onValueChange(event: unknown): void {
+        event;
+        if (this.courseTimeZone === '' || this.courseTimeZone === DateTimeFieldComponent.localTimeZone) {
+            this.courseTime = undefined;
+            return;
+        }
+        if (!this.isTimeValid || !isValid(this.value)) {
+            this.courseTime = 'Course Time: ';
+        } else {
+            this.courseTime = `Course Time: ${formatInTimeZone(
+                this.value,
+                this.courseTimeZone,
+                'MMM dd, yyyy HH:mm:ss'
+            )}`;
+        }
+    }
+
+    public override set srcValue(srcValue: Date | [Date, string]) {
+        if (Array.isArray(srcValue)) {
+            this.value = srcValue[0];
+            this.courseTimeZone = srcValue[1];
+        } else {
+            this.value = srcValue;
+            this.courseTimeZone = '';
+        }
+        this.onValueChange({});
```

Comment:
Actually, `bsValueChange` and `isValid` events are fired in the next Angular change detection cycle instead of during the assignment at line `68` or line `71`. This is because the Angular property binding (line `9` and line `16` of `date-time-field.component.html`) won't synchronously propagate changes. They wait until the next Angular change detection cycle which re-evaluates expressions for the template.

As a result, the initialization of `courseTimeZone` will happen before `onValueChange()` being invoked. You can verify this by adding `console.log()` or `console.trace()` to the `srcValue` setter and the `onValueChange()` method.

Again, this invocation of `onValueChange()` (renamed to `updateCourseTimeText()` now) is not necessary but will not cause unexpected result.